### DISCUSSION TITLE
Is there any way that vllm works with huggingface pipeline?

### DISCUSSION THREAD

**Discussion by linkfqy** [vllm-project/vllm#3761] [Q&A]
Since `vllm.LLM` is not subclass of `transformers.PretrainedModel`, I can't use vllm like `pipeline(model=vllm_model)`.
My code is deeply coupled with Pipeline, and modifying multiple places to introduce vllm is troublesome for me.

**Comment by yileitu** [comment]
Same here. I need to use 
```python
pipeline = transformers.pipeline(
    "text-generation",
    model=model,
    tokenizer=tokenizer,
)
```
